### ISSUE TITLE
Changeset source is wrong when imagery is not Bing

### ISSUE THREAD

**Issue by pyrog** [hotosm/osm-tasking-manager2#530]
When editing a task with JOSM, the source is the imagery URL but should be the value defined by project managers.

**Exemple**
[#837 - Typhoon Ruby (Hagupit) - Eastern Samar Post-Disaster UAV Mapping](http://tasks.hotosm.org/project/837#task/39)

![project 837 imagery](https://cloud.githubusercontent.com/assets/2158081/5758478/3d3aeec8-9cc6-11e4-9342-5974d4e208a8.png)
![project 837 source](https://cloud.githubusercontent.com/assets/2158081/5758484/416faa2e-9cc6-11e4-9d2b-b67f0341e991.png)

There is no field "source" in the project so it couldn't be passed automatically to JOSM.
I suggest to add it in the image tab of project editing page:

![project editing](https://cloud.githubusercontent.com/assets/2158081/5758778/89c44bca-9cc8-11e4-93ef-b9eb7e8603ff.png)

**Comment by bgirardot** [comment]
@pyrog Is this issue the same one where I get the TMS url in the JOSM source field automatically?

For example:
![2015-01-15_15h51_10](https://cloud.githubusercontent.com/assets/265831/5759467/6c6585c0-9cce-11e4-8719-a39e487a8135.png)

Instead of:
![2015-01-15_15h51_22](https://cloud.githubusercontent.com/assets/265831/5759476/76155618-9cce-11e4-8409-c660ab09ac95.png)


Because if it is, I felt it was a bug when I first saw it and it might be a bug, but to be hones, the "bug"ed behavior is really a lot more useful than the "proper" behavior.

I am often looking up what other people have done and see some source like "WorldView September 14th 2012" and that means nothing to me, how can I view that imagery to see if it really shows what has been mapped in?

I have to google search the string and hope I find the tm job or the wiki page that then might have the tms url I can use in JOSM.

Putting the tms url direction into the changeset tags is a more useful to me, I can just grab it, load it in JOSM and see if the building really is shaped like the letter S :)

So now I am tending to leave the "wrong" source in. it is not really even wrong, it is just very super specific which is great for me.

If anything we should be adding another changset tag besides source like "imagery_url" or the now famous "imagery_used" field iD puts in and source stays the semi-human readable description to properly credit the image donors, which is very important. But equally important is that I can review the imagery if possible in the future to confirm the mapping matches it.

**Comment by pierzen** [comment]
Programming of the Task Manager have to be taught at every step to let cthe Managers control the contents adapted to various situations, provide them flexibiility. 

We should keep the source field description independant of the imagery. There are tasks where many inputs are used. For example, to import Place Names, we used simultaneously Bing, Topo Maps, GNS.

+ 1 for the source field in the Imagery panel. I would call this Source Description

**Comment by bgirardot** [comment]
Oh I agree 100% @pierzen I guess my main point was just that for these examples of "one off" imagery I'd really like to have the url on the changsets so i can find it in the future when reviewing work in whatever way we could put it on the changeset.

But I will stop letting the url go in source field, upon reflection, that is a mistake on my part, our partners who provide this imagery need to be credited properly first and foremost.

**Comment by pierzen** [comment]
Yes the Imagery providers need to be credited and the Activators in particular, we take care of this, adding the appropriate source description. But we need flexibility, no automatic content.  For cases where **Imagery Source description** is provided in the **Imagery panel** and no **Data source** is provided in the **Settings Pannel**, then the Imagery Source Description could be used.

**Comment by bgirardot** [comment]
I don't think we disagree @pierzen I am just saying, if there is a custom imagery url in the tasking manager, I would like to include a key=value pair on the changeset for imagery_url=[whatever].

I understand it is not going to cover jobs where I might load other imagery layers like a GNS import, but for the vast majority of jobs, I just want the "one off" imagery url we are using for mapping, like all the hiu imagery urls we often have on projects.

And source gets fixed so it works the way it used to, it does not work correctly now in JOSM, it is not passing the source field set in the tasking manager by the project manager.

Now we have to manually copy/paste the value for the source field in JOSM.

**Comment by pyrog** [comment]
@pierzen
I don't see a **Settings Pannel**, do you mean **Instructions" tab ?

![instructions tab](https://cloud.githubusercontent.com/assets/2158081/5777103/ea689d60-9d8f-11e4-94ec-8edf4278d40f.png)

In fact there are 3 places where we could add a **Changeset Source** field:
* In the **Instruction** tab
* In the **imagery** tab
* In the **Edit licence** page but we need to review more things

@bgirardot 
We could discuss with JOSM dev how we could pass the image_url in the changeset metadatas.
And another important thing, if we could pass the **[Imagery Offset](http://learnosm.org/en/editing/correcting-imagery-offset/)**

**Comment by pierzen** [comment]
@pyrog  Sorry I interpreted @bgirardot ticket and did not realized that the panel example was from JOSM.  With the constant modifications to the panels introduced through tickets without prior discussions on the overall TM design, it is becoming difficult to follow even for intensive TM editors like me.

So where are we now exactly with this source field? Coordinators want some flexibility, Image providers like to mention of their contribution and we want to pass this info to JOSM. And as I said yesterday, there could be more then one imagery used. 

To keep this simple, I suggest to add to the Imagery panel a Imagery source field. This field could be showed

Imagery source (s)   
Note: for projects using many sources you can write for example (Bing; GNS; JOG Topo map)
+------------------------------------------------------------------+
|                                                                                    |
+------------------------------------------------------------------+

**Comment by pyrog** [comment]
> we want to pass this info to JOSM

In fact, all OSM users want/need a Changeset source :wink:

So this is the first "screen copy" at the beginning of this discussion (I added this big red arrow for you) :wink: 

> To keep this simple, I suggest to add to the Imagery panel a Imagery source field.

![imagery source](https://cloud.githubusercontent.com/assets/2158081/5780910/8fe449b0-9dae-11e4-9843-4686a18dfc03.png)

**Comment by ethan-nelson** [comment]
Relevant area of code is: https://github.com/hotosm/osm-tasking-manager2/blob/master/osmtm/static/js/project.js#L384

We could add another field on the imagery page, but that will require a database script (which I don't even know where to begin). 

The JS script would probably then:
* First check if that custom title field is not blank
* Then check if the url is not blank
* Default to Bing otherwise

**Comment by danbjoseph** [comment]
+1 to this. when the changeset metadata has the source as the url (e.g.`Custom (http://api.tiles.mapbox.com/v3/americanredcross.mcb248ih/{z}/{x}/{y}.png)`) this is not very helpful or descriptive.
